xds/internal/balancer/clusterresolver/resource_resolver_dns.go

```diff
@@ -142,7 +142,17 @@ func (dr *dnsDiscoveryMechanism) ReportError(err error) {
 		dr.logger.Infof("DNS discovery mechanism for resource %q reported error: %v", dr.target, err)
 	}
 
-	dr.topLevelResolver.onError(err)
+	dr.mu.Lock()
+	// Don't report any errors after first error or good address update.
+	if dr.addrs != nil {
+		dr.mu.Unlock()
+		return
+	}
+	dr.addrs = make([]string, 0)
+	dr.updateReceived = true
+	dr.mu.Unlock()
+
+	dr.topLevelResolver.onUpdate()
```

Comment:
Since we are calling `onUpdate` for both good updates and errors, we no longer need the `onError` method on the `topLevelResolver`. EDS discovery mechanism does not use it either. So, you can delete the implementation of that method in `resourceResolver` and remove that method from the `topLevelResolver` interface.

Reply:
Hmmm ok. Will delete.

Reply:
It's actually being called in DNS Discovery Mechanism Construction errors, but as per the back and forth in chat I need to switch this to do same thing DNS errors before update do - create the priority and send TF down.

Reply:
I did this in Discovery Mechanism Constructor same way as OnError - set updateReceived to true and then call onUpdate on the top level resolver. Let me know if you want to change the bool (I need to set it for lastUpdate() to return true).

Reply:
> Let me know if you want to change the bool

What do you mean by this? I'm good with how it is looking now.

Reply:
I changed the construction to also set this bool and trigger TF. Thus, I don't know if the naming is appropriate. Has it received an update, or is it just triggering an error? I'm fine leaving as is also though.